owlbot-postprocessor/lib/owlbot/common_modifiers.rb

```diff
@@ -134,6 +134,26 @@ def prevent_overwrite_of_existing path, name: nil
       end
     end
 
+    ##
+    # A temporary convenience method that adds `ostruct` as required for test helpers files,
+    # which are located in `test/helper.rb`.
+    #
+    # This is only added for auto-generated files that do not have the dependency explicitly listed.
+    #
+    # @param name [String] Optional name for the modifier to add. Otherwise, a default is provided.
+    #
+    def require_open_struct_in_test_helper name: nil
+      path = [/test\/helper\.rb$/]
+      name ||= "require_open_struct_in_test_helper"
+      modifier path: path, name: name do |src, _dest|
+        if src&.match("Auto-generated by gapic-generator-ruby.") && !src&.match?(/require "ostruct"/)
+          src += 'require "ostruct"'
+          src += "\n"
```

Comment:
Consider:

```ruby
src = "#{src}\nrequire \"ostruct\"\n"
```

This creates a new string rather than modify the existing one in place, which is usually safer. (In this case, the postprocessor framework guards against any issues with doing so, but it's good practice not to mutate existing objects anyway.) Also, it's safer to insert a newline before the line you append, just in case the original file didn't end with a newline. (Again, in this case, the existing files should be well-behaved in practice, but better to code defensively.)

Reply:
Thanks! I should have used the double quotes, and then escaped the other ones (I noticed '\n' doesn't work).